tools/sense_studio/templates/testing.html

```diff
@@ -0,0 +1,124 @@
+                    <select class="uk-select {{ 'uk-form-danger' if not classifiers }}" id="classifier">
+                        {% if classifiers %}
+                            {% for classifier in classifiers %}
+                                <option>{{ classifier }}</option>
+                            {% endfor %}
+                        {% else %}
+                            <option value="">No classifiers to select from</option>
+                        {% endif %}
+                    </select>
+                </div>
+            </div>
+
+            <div class="uk-hidden">
+                <label class="uk-form-label" for="title">Window Title</label>
+                <div class="uk-form-controls">
+                    <div class="uk-width-1-1 uk-inline">
+                        <span class="uk-form-icon" uk-icon="icon: pencil"></span>
+                        <input class="uk-input" type="text" id="title" placeholder="Title" autocomplete="off" />
+                    </div>
+                </div>
+            </div>
+
+            <div>
+                <label class="uk-form-label">Input Source</label>
+                 <div class="uk-form-controls">
+                     <div class="uk-inline uk-width-1-1">
+                        <label>
+                            <input class="uk-radio" type="radio" name="inputSource" value="0"
+                                   autocomplete="off" onclick="toggleInputVideoField();" checked>
+                            Webcam
+                        </label>
+                        <label class="uk-margin-large-left">
+                            <input class="uk-radio" type="radio" name="inputSource" value="1"
+                                   autocomplete="off" onclick="toggleInputVideoField();">
+                            Input Video
+                        </label>
+                     </div>
+                </div>
+            </div>
+
+            <div id="inputVideoDiv" class="uk-hidden">
+                <label class="uk-form-label" for="inputVideoPath">Input video path <span class="uk-label">.mp4</span></label>
+                <label class="uk-form-label uk-text-danger" id="inputVideoPathLabel"></label>
+                <div class="uk-form-controls">
+                    <div class="uk-inline uk-width-1-1">
+                        <span class="uk-form-icon" uk-icon="icon: video-camera"></span>
+                        <input class="uk-input file-path-search" type="text" id="inputVideoPath" placeholder="/path/to/input-video"
+                               autocomplete="off" oninput="editInputVideoPath();">
+                    </div>
+                </div>
+            </div>
+
+            <div>
+                 <div class="uk-form-controls uk-width-1-1">
+                    <input class="uk-checkbox" type="checkbox" id="saveVideo" autocomplete="off"
+                           onclick="toggleOutputVideoField();">
+                        Save Output Video
```

Comment:
Neither the `Input Video` nor the `Save Output Video` field force you to actually enter the required information. So it is possible to check these boxes and start a training, which will ignore those settings. It is even possible to start a training with an incomplete or wrong input video path, which will just stop without any visible error message.
I think this is the final missing detail on this page.

Reply:
did updates to handle  these situations